### ISSUE TITLE
kwin becomes unresponsive, upstream bug?

### ISSUE THREAD

**Issue by turion** [NixOS/nixpkgs#206868]
When using Alt + Tab often, all windows freeze.

This seems to be a known upstream bug: https://bugs.kde.org/show_bug.cgi?id=457847#c22 Nevertheless I wanted to report here for documentation and to simplify debugging for other people.

A temporary workaround seems to be to restart `kwin` from a terminal (use Ctrl+Alt+F6 to get there):

```
systemctl --user restart plasma-kwin_x11
```

The underlying issue is probably a buggy version of mesa, 22.2. When downgrading mesa with an overlay like this, it should work:

```
            nixpkgs.overlays = [(self: super: {
              inherit (import nixpkgs-22-05 { inherit system; })
                mesa
              ;
            })];
```

A new version of mesa will be released before the end of the year, containing the fix. This issue here can be closed when the corresponding version bump has landed in nixpkgs.

**Comment by vcunat** [comment]
With mesa-22.2.5 it's still broken?  I wouldn't want to switch mesa's major version from 22.2 on stable NixOS 22.11.

**Comment by turion** [comment]
> With mesa-22.2.5 it's still broken?

As far as I understand yes. Further upstream, they assert that 22.3.0 will contain the fix: https://gitlab.freedesktop.org/mesa/mesa/-/issues/7674#note_1680853

I have to add that I'm not completely convinced yet that this is the whole story of the bug, because I have two Lenovo ThinkPads with nearly identical config, and on one I experience the bug and on the other one, not. This may be because I'm using one of them more, though. Long story short, it would be good if other people on Nixos can confirm they experience this bug on up-to-date nixos-unstable or nixos-22.11 and KDE.

**Comment by tobiasBora** [comment]
I actually confirm this bug, since my recent NixOs upgrade my system crashes (often, the applications are still usable, but alt-tab to switch does not work anymore, and I can't change window without killing the windows that are above (typically I try to open a new terminal, then I kill the apps from the terminal, then I kill the terminal…). While it is handy to save work, before restarting the OS, it's really annoying… and I actually nearly experienced work loses as sometimes the input system also stops working.

The issue I was looking at is https://bugs.kde.org/show_bug.cgi?id=461316 and the fix is merged here https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/19972

I checked and the fix is NOT in 22.2.5 (currently in use in NixOs), but is present in all the 22.3.x series (the latest version is 22.3.3 as of today). Would it be possible to upgrade mesa please?

@vcunat  @primeos

**Comment by turion** [comment]
The fix is in https://github.com/NixOS/nixpkgs/pull/204125, which was merged into `nixos-unstable`.

**Comment by turion** [comment]
Let's both test whether it works after the fix, and if yes, close the issue.

**Comment by tobiasBora** [comment]
Yes ! After a day of use I had no issue so far. I don't know however if it needs backport. Technically the problematic commit was 6 months old, and mesa 22.2.4, in use on NixOs 22.11, is like two months old…

**Comment by vcunat** [comment]
I glanced at the commits quickly, and the culprit is reachable from all 22.2.x and doesn't appear reverted on the branch yet.  (and it's been declared unmaintained in the meantime)  Well, maybe it's possible to simply pick the revert pair that landed in 23.0.0, but I don't know the code-base.

**Comment by tobiasBora** [comment]
Wouldn't it be simpler to just upgrade mesa like in unstable?

**Comment by vcunat** [comment]
Well, if I knew that it fixes much more users than it breaks...
There are reasons why larger upgrades are generally avoided on stable.

**Comment by tobiasBora** [comment]
Yeah, make sense.
